components/mpas-ocean/src/shared/mpas_ocn_diagnostics.F

```diff
@@ -667,6 +668,152 @@ subroutine ocn_relativeVorticity_circulation(relativeVorticity, circulation, nor
+!  routine ocn_diagnostic_solve_wctEdge
+!
+!> \brief   Computes layer thickness at edge
+!> \author  Carolyn Begeman
+!> \date    July 2023
+!> \details
+!>  This routine computes the water column thickness on edges, wctEdge, which 
+!>  can either be the mean or an upwind value, depending on user input.
+!
+!-----------------------------------------------------------------------
+
+   subroutine ocn_diagnostic_solve_wctEdge(btrVelocity, sshCell, &
+                         bottomDepthEdge, wctEdge)!{{{
+
+      !-----------------------------------------------------------------
+      ! input variables
+      !-----------------------------------------------------------------
+
+      real (kind=RKIND), dimension(:), intent(in) :: &
+         btrVelocity,   &!< [in] barotropic velocity
+         sshCell,       &!< ssh at cells, used for centered water column
+                         !< thickness
+         bottomDepthEdge !< bottom depth centered at edges, used for
+                         !< centered water column thickness
+      ! bottomDepth is already present
+
+      !-----------------------------------------------------------------
+      ! output variables
+      !-----------------------------------------------------------------
+
+      ! Outputs are the shared variables from diagnostic_variables:
+      real (kind=RKIND), dimension(:), intent(inout) :: &
+         wctEdge !< [out] water column thickness on edge
+
+      !-----------------------------------------------------------------
+      ! local variables
+      !-----------------------------------------------------------------
+
+      integer :: &
+         iCell, iEdge, &! edge and cell loop indices
+         cell1, cell2  ! neighbor cell indices across an edge
+
+      real (kind=RKIND), dimension(:), allocatable :: wctCell, sshEdge
+
+      ! End preamble
+      !-----------------------------------------------------------------
+      ! Begin code
+
+      wctEdge(:) = -1.0e34_RKIND
+      ! Compute edge flux based on option set on init
+      select case (thickEdgeFluxChoice)
+
+      case (thickEdgeFluxCenter)
+         ! Use centered (mean) thickness as flux value
+
+         ! Compute mean layer thickness at edge
+#ifdef MPAS_OPENACC
+         !$acc enter data create(bottomDepthEdge, sshCell, sshEdge, wctEdge)
+         !$acc parallel loop &
+         !$acc    present(bottomDepthEdge, sshCell, sshEdge, wctEdge, cellsOnEdge) &
+         !$acc    private(iEdge, cell1, cell2)
+#else
+         !$omp parallel
+         !$omp do schedule(runtime) private (cell1, cell2)
+#endif
+         do iEdge = 1, nEdgesAll
+            cell1 = cellsOnEdge(1,iEdge)
+            cell2 = cellsOnEdge(2,iEdge)
+            wctEdge(iEdge) = 0.5_RKIND*(sshCell(cell1) + sshCell(cell2)) &
+                             + bottomDepthEdge(iEdge)
+         end do
+#ifndef MPAS_OPENACC
+         !$omp end do
+         !$omp end parallel
+#endif
+
+#ifdef MPAS_OPENACC
+         !$acc update host(wctEdge)
```

Comment:
use copyout instead of update/delete for wctEdge as in:

!$acc exit data delete(bottomDepthEdge, sshCell, sshEdge) &
!$acc                 copyout(wctEdge)